src/hotspot/cpu/x86/x86.ad

```diff
@@ -9166,8 +9345,39 @@ instruct mask_all_evexL(kReg dst, rRegL src) %{
   format %{ "mask_all_evexL $dst, $src \t! mask all operation" %}
   ins_encode %{
     int vec_len = Matcher::vector_length(this);
-    __ kmovql($dst$$KRegister, $src$$Register);
-    __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
+    if (VM_Version::supports_avx512bw()) {
+      __ kmovql($dst$$KRegister, $src$$Register);
+      __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
+    } else {
+      assert(vec_len <= 16, "");
+      __ kmovwl($dst$$KRegister, $src$$Register);
+      __ kshiftrwl($dst$$KRegister, $dst$$KRegister, 16 - vec_len);
+    }
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct mask_not_immLT8(kReg dst, kReg src, rRegI rtmp, kReg ktmp, immI_M1 cnt) %{
```

Comment:
What happens to not operation if vector_length < 8 and it is a !avx512dq platform?

Reply:
In that case this pattern will not get matched since instruction sequence will still be same.

Following two implementations are possible for targets without DQ:
A) XorVMask  SRC (MaskAll -1):
  Instruction sequence
  KMOV -1 SRC1
  KSHIFTRL 16-masklen, SRC1
  KXOR SRC1, DST
B) New instruction for NOT operation.
In this case following will be generated instruction sequence
KNOTW SRC1
KMOV  FILTER 3/15  
KAND SRC FILTER

Apparently there is not much difference in the two implementations.